src/node/desktop/src/core/xdg.ts

```diff
@@ -0,0 +1,293 @@
+/*
+ * xdg.ts
+ *
+ * Copyright (C) 2021 by RStudio, PBC
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+/*
+ * These routines return system and user paths for RStudio configuration and data, roughly in
+ * accordance with the FreeDesktop XDG Base Directory Specification.
+ *
+ * https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html
+ *
+ * All of these can be configured with environment variables as described below.
+ *
+ * The values of the environment variables can include the following special variables:
+ *
+ * $USER  The user's name
+ * $HOME  The user's home directory
+ * ~      The user's home directory
+ *
+ * These values will be resolved against the current user by default. If you wish to
+ * resolve them against a different user, supply their name and home directory using
+ * the boost::optional arguments.
+ */
+
+import os from 'os';
+import { Environment, expandEnvVars, getenv, setenv } from './environment';
+import { username, userHomePath } from './system';
+import { FilePath } from './file-path';
+
+export enum WinFolderID {
+  FOLDERID_RoamingAppData,
+  FOLDERID_LocalAppData,
+  FOLDERID_ProgramData
+}
+
+/**
+ * Simplified implementation of Windows `SHGetKnownFolderPath` API using environment
+ * variables.
+ */
+export function SHGetKnownFolderPath(folderId: WinFolderID): string {
+  let envVar = "";
```

Comment:
This is ultra bike-shedding, but should we prefer using single-quoted strings or double-quoted strings?

Reply:
There's no universal preference I've found. Some projects prefer single, some double, and of course if you want to include embedded quotes in a literal then easiest to use the other one around it.

We can pick one and configure the tools to auto-format it that way if we want, along with other conventions.